stdlib/public/core/Collection.swift

```diff
@@ -234,17 +234,19 @@ extension CollectionType where SubSequence == Self {
     self = self[startIndex.successor()..<endIndex]
     return element
   }
+}
 
+extension CollectionType where
+    SubSequence == Self, Index: BidirectionalIndexType {
```

Comment:
I'm afraid we don't have it written down.  Please imitate the coding style of the code you see in the library.